### PR TITLE
Sample FreeRTOSConfig.h and template port

### PR DESCRIPTION
**Pull request by n9wxu**
Sample FreeRTOSConfig.h and template port

Description
-----------
This change adds a FreeRTOSConfig.h with all features documented.  This configuration does compile but is intended to be a starter file for your application and is not guaranteed to have the most appropriate settings for you.  Additionally, this configuration of this file is not guaranteed to be stable over time.

Additionally, this push adds a "template" port called template.  This port is intended to allow the kernel to compile but it will not create a functional RTOS.  The purpose of this port is as follows:
1) Provide a starting point for future ports with the minimum functions needed by the port.
2) Provide a compilable "port" to support executing static analysis automation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,7 +121,11 @@ See the readme file in the ```./portable``` directory for more information.
 
 - The ```./include``` directory contains the real time kernel header files.
 
+- The ```./sample_configuration``` directory contains sample a `FreeRTOSConfig.h` to help jumpstart a new project.
+See the `FreeRTOSConfig.h` file for instructions.
```
Comment (AniruddhaKanhere) [inline README.md:125]: We can make this into a link by giving a relative path like so:

[MyFile.h](portable/myDirectory/MyFile.h)

This is something which GitHub understands AFAIK.

File: sample_configuration/FreeRTOSConfig.h [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+#define INCLUDE_vTaskDelay                     1
+#define INCLUDE_xTaskGetSchedulerState         1
+#define INCLUDE_xTaskGetCurrentTaskHandle      1
+#define INCLUDE_uxTaskGetStackHighWaterMark    0
+#define INCLUDE_xTaskGetIdleTaskHandle         0
+#define INCLUDE_eTaskGetState                  0
+#define INCLUDE_xEventGroupSetBitFromISR       1
+#define INCLUDE_xTimerPendFunctionCall         0
+#define INCLUDE_xTaskAbortDelay                0
+#define INCLUDE_xTaskGetHandle                 0
```
Comment (archigup) [inline sample_configuration/FreeRTOSConfig.h:424]:
```suggestion
#define INCLUDE_vTaskPrioritySet               1
#define INCLUDE_uxTaskPriorityGet              1
#define INCLUDE_vTaskDelete                    1
#define INCLUDE_vTaskSuspend                   1
#define INCLUDE_xResumeFromISR                 1
#define INCLUDE_vTaskDelayUntil                1
#define INCLUDE_vTaskDelay                     1
#define INCLUDE_xTaskGetSchedulerState         1
#define INCLUDE_xTaskGetCurrentTaskHandle      1
#define INCLUDE_uxTaskGetStackHighWaterMark    1
#define INCLUDE_xTaskGetIdleTaskHandle         1
#define INCLUDE_eTaskGetState                  1
#define INCLUDE_xEventGroupSetBitFromISR       1
#define INCLUDE_xTimerPendFunctionCall         1
#define INCLUDE_xTaskAbortDelay                1
#define INCLUDE_xTaskGetHandle                 1
```

**Review by Skptak — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

**Pull request merged**